src/backend/joanie/core/models/products.py

```diff
@@ -397,6 +421,58 @@ def can_edit(self):
         """Return True if the order group can be edited."""
         return not self.orders.exists()
 
+    @property
+    def available_seats(self) -> int | None:
+        """Return the number of available seats on the order group."""
+        if self.nb_seats is None:
+            return None
+        return self.nb_seats - self.get_nb_binding_orders()
+
+    # ruff: noqa: PLR0911
+    # pylint: disable=too-many-return-statements
+    @property
+    def is_enabled(self):
+        """
+        Returns boolean whether the order group is active based on its status, availability,
+        and time constraints.
+        When a start or end date is setted, we need to verify if we are in the period
+        of early bird, or last minute depending on the course run enrollment start or
+        enrollment end datetimes. When both dates are setted, we need to make sure that the
+        current day is between those.
+        """
+        if not self.is_active:
+            return False
+        # Unlimited spots
+        if self.nb_seats is None:
+            return True
+        if self.nb_seats - self.get_nb_binding_orders() == 0:
+            return False
+        if not self.start and not self.end:
+            return True
+
+        now = timezone.now()
+        if self.start and self.end:
+            return self.start <= now <= self.end
+
+        if self.course_product_relation.product.type == enums.PRODUCT_TYPE_CERTIFICATE:  # pylint: disable=no-member
+            course_run_dates = (
+                self.course_product_relation.course.get_equivalent_course_run_dates(  # pylint: disable=no-member
+                    ignore_archived=True
+                )
+            )
+        else:
+            course_run_dates = (
+                self.course_product_relation.product.get_equivalent_course_run_dates(  # pylint: disable=no-member
+                    ignore_archived=True
+                )
+            )
```

Comment:
I feel we could move that on the `CourseProductRelation` model.

```suggestion
self.course_product_relation.get_equivalent_course_run_dates(ignored_archived=True)
```

Reply:
This method is not implemented into the course product relation model.
```
*** AttributeError: 'CourseProductRelation' object has no attribute 'get_equivalent_course_run_date'
```
Do you want me to add it ? or keep the way it's actually done in `is_enabled` property ?